Automattic/jetpack
# SSO: disable complete wpcom invite interface and extract utility

> [!NOTE]
> ~This is a proposal for a change to #36549.~
>  This is an alternative to #36549.

## Proposed changes:

PR #35902 introduced the new "Invite user to WordPress.com" checkbox that queries WPCOM API via user token.
The checkbox is displayed for all users, but those non-connected don't have a user token, so they aren't able to add new WordPress user accounts with the checkbox checked.

- Hide the "Invite user to WordPress.com" checkbox for non-connected users.
- Disable the invitation sending functionality for non-connected users.

This also introduces a new filter, `jetpack_sso_invite_new_users_wpcom`, allowing third-parties to completely disable that interface.

### Other information:


## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* On a new site, connect only the site to WordPress.com.
* Go to Users > Add New
    * You should not see any WordPress.com invite component.
* Go to Jetpack > Dashboard and connect your user account as well
* Go to Users > Add New
    * The wpcom invite interface should now appear.
* In a functionality plugin, add the following: `add_filter( 'jetpack_sso_invite_new_users_wpcom', '__return_false' );`
* Go back to Users > Add New
    * The wpcom invite interface should now be gone.
